vllm/config/model.py

```diff
@@ -466,9 +470,18 @@ def __post_init__(
             # We'll determine how to apply dict overrides after loading the config
             hf_overrides_kw = {}
             dict_overrides = {}
+            if self.quantization_schema:
+                quant_config_override = get_default_quantization_hf_config(
+                    self.quantization, self.quantization_schema
+                )
+                if quant_config_override:
+                    dict_overrides["quantization_config"] = quant_config_override
             for key, value in self.hf_overrides.items():
                 if isinstance(value, dict):
-                    dict_overrides[key] = value
+                    if dict_overrides.get(key):
+                        dict_overrides[key] = {**dict_overrides[key], **value}
```

Comment:
this seems rather convoluted. 

We're assuming dict_overrides[key] is always a mapping as long as dict_overrides.get(key) and isinstance(value, dict) are true which doesn't seem obvious

as far as the logic:

we take the dict that we assume we get from dict_overrides[key] and add the value dict to it, letting value take precedence where there are any conflicts.

Feels like we should be taking one or the other i.e. overriding things rather than this merge operation.

also still no warning when one thing overrides another.